internal/renderer/renderer.go

```diff
@@ -51,6 +51,21 @@ var (
 // matches a FULL string that contains no non-whitespace characters
 var emptyLineRegex = regexp.MustCompile(`\A[\s]*\z`)
 
+// fairly tolerant to handle weird HTML
+var tagPairRegexString = `<[\n\f ]*%s([\n\f ]+[^\n\f \/>"'=]+[\n\f ]*(=[\n\f ]*([a-zA-Z1-9\-]+|"[^\n\f"]+"|'[^\n\f']+'))?)*[\n\f ]*>.*?<[\n\f ]*/[\n\f ]*%s[\n\f ]*>`
+
+// HTML block tags whose contents should not be rendered
+var htmlNoRenderRegex = []*regexp.Regexp{
+	regexp.MustCompile(fmt.Sprintf(tagPairRegexString, "fieldset", "fieldset")),
+	regexp.MustCompile(fmt.Sprintf(tagPairRegexString, "form", "form")),
+	regexp.MustCompile(fmt.Sprintf(tagPairRegexString, "iframe", "iframe")),
```

Comment:
What about older tags such as `frame`? While technically unsupported in HTML5 they still work in browsers. One would have to do that on purpose, of course (but so one would to have a Web form inside a Markdown file).

There's also `noscript`, which is presented to users with JavaScript disabled. The logic to whether or not to render its contents at all is debatable.

PS: yeah, handling Web is "fun".

Reply:
Ha, yeah I could dig into the MDN docs all day I'm sure and find tags I never knew existed. I looked at the list of tags here since it's all that gomarkdown looks for: https://github.com/gomarkdown/markdown/blob/master/parser/block.go

I almost included noscript! I'm not sure either. Gemini doesn't support JavaScript, so I guess "noscript" would be appropriate? The only time I can imagine someone putting this into Markdown is when they have some kind of interactive JS thing on the page and they want to tell people that it can't be viewed. As a placeholder that's probably fine.